controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/screwdriver/ScrewdriverApiTest.java

```diff
@@ -57,12 +61,32 @@ public void testTriggerJobForApplication() {
                                    app.id().tenant().value() + "/application/" + app.id().application().value(),
                                    new byte[0], Request.Method.POST),
                        200, "{\"message\":\"Triggered component for tenant1.application1\"}");
+        tester.controller().applications().deploymentTrigger().notifyOfCompletion(new JobReport(app.id(),
+                                                                                                DeploymentJobs.JobType.component,
+                                                                                                1,
+                                                                                                42,
+                                                                                                Optional.of(new SourceRevision("repo", "branch", "commit")),
+                                                                                                Optional.empty()));
 
-        // Triggers specific job when given -- fails here because the job has never run before, and so application version can't be resolved.
+        // Triggers specific job when given -- triggers the prerequisites here, since they are not yet fulfilled.
```

Comment:
The intention behind this API was that it should trigger given job unconditionally (just a proxy for triggering a job on Screwdriver). It's a tool for exceptional cases, where e.g. orchestration does not work as intended and manual intervention is needed.

Reply:
Staging deployment would also fail with 4XX if the test wasn't through system yet. Now this API triggers system instead, if it isn't already a success, and you try to trigger staging.

Reply:
Actually, there's a real reason for this: since deployment targets may be different for different production zones, it is necessary to set the test job targets _here_. An alternative could be to trigger with given version arguments, but that seems more clumsy, in my opinion.